Automattic/wp-calypso
# Site Editor: Remove unused site editor feedback form logic

#### Changes proposed in this Pull Request

Note: This PR is dependent on https://github.com/Automattic/wp-calypso/pull/62357 being deployed first.

The site editor feedback form is no longer necessary. We remove the UI for the site editor feedback form [here](https://github.com/Automattic/wp-calypso/pull/62357). In this PR, we remove the underlying logic supporting the feedback form plugin as well as its accompanying css.

#### Testing instructions



- Checkout branch
- Sandbox widgets.wp.com
- Sync wpcom-block-editor changes to sandbox PCYsg-l4k-p2
- Navigate to `calypso/apps/wpcom-block-editor` in terminal
- Execute `yarn dev --sync`
- Smoke test the site editor and verify that it continues to work as expected



Related to https://github.com/Automattic/wp-calypso/issues/61774 and https://github.com/Automattic/wp-calypso/pull/62357
